backend/src/main/java/hanglog/image/service/ImageService.java

```diff
@@ -0,0 +1,111 @@
+package hanglog.image.service;
+
+import static hanglog.global.exception.ExceptionCode.IMAGE_LIST_EMPTY;
+import static hanglog.global.exception.ExceptionCode.IMAGE_LIST_SIZE_EXCEED;
+import static hanglog.global.exception.ExceptionCode.IMAGE_NULL;
+import static hanglog.global.exception.ExceptionCode.IMAGE_PATH_INVALID;
+import static org.springframework.util.StringUtils.getFilenameExtension;
+
+import hanglog.global.exception.ImageException;
+import hanglog.image.dto.ImagesResponse;
+import hanglog.image.util.ImageNameUrlConverter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.LocalDateTime;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+@Service
+public class ImageService {
+
+    private static final String EXTENSION_DELIMITER = ".";
+    private static final int MAX_IMAGE_LIST_SIZE = 10;
+    private static final int EMPTY_LIST_SIZE = 0;
+
+    @Value("${image.base-path}")
+    private String imageBasePath;
+
+    public ImagesResponse save(final List<MultipartFile> images) {
+        validateImagesSize(images);
+        validateImageNull(images);
+
+        makeDirectoryIfNotExist();
+
+        final List<String> hashedNames = images.stream()
+                .map(this::hashName)
+                .toList();
+
+        final List<Path> imagePaths = hashedNames.stream()
+                .map(hashedName -> Path.of(imageBasePath, hashedName))
+                .toList();
+
+        uploadImages(images, imagePaths);
+
+        final List<String> imageUrls = hashedNames.stream()
+                .map(ImageNameUrlConverter::convertNameToUrl)
+                .toList();
+
+        return new ImagesResponse(imageUrls);
+    }
+
+    private void validateImageNull(final List<MultipartFile> images) {
+        images.forEach(image -> {
+            if (image.isEmpty()) {
+                throw new ImageException(IMAGE_NULL);
+            }
+        });
+    }
```

Comment:
```suggestion
    private void validateImageNull(final List<MultipartFile> images) {
        final boolean hasNullImage = images.stream()
                        .anyMatch(MultipartFile::isEmpty);
        if (hasNullImage) {
            throw new ImageException(IMAGE_NULL);
        }
    }
```

이런 방식도 있는데 길이는 똑같네요 ..

Reply:
반영완!